awx/main/models/credential/__init__.py

```diff
@@ -504,6 +508,15 @@ class Meta:
                     'Ansible Tower documentation for example syntax.')
     )
 
+    @classmethod
+    def from_db(cls, db, field_names, values):
+        instance = super(CredentialType, cls).from_db(db, field_names, values)
```

Comment:
If `managed_by_tower` and `native_source` are set, we grab `inputs` and `injectors` from code (the database just has `{}` for these).

Reply:
I had a hard time understanding why `native_source` is necessary at all, but I _think_ it appears to have the same meaning as the old v1 kind.

Reply:
We need _some_ static, unique identifier to map the database row to a block of code (the namespace defined in the registry).  This will be especially important w/ @jakemcdermott's work where _other Python packages_ could be imported that define these sources.

Reply:
I would actually go further, users could really use this to look up credential types. If you look at how tower-cli did it:

```
tower-cli credential list --credential-type="Amazon Web Services"
```

People would probably prefer to filter by "aws", and client libraries right now will go out of their way to create a mapping between the two, or worse, use v1.

Reply:
I could _maybe_ see us exposing this identifier in the API and letting people filter on it.